test/api/model/model_test.dart

```diff
@@ -628,4 +631,100 @@ void main() {
         .deepEquals([2, 3, 11]);
     });
   });
+
+  group('MessageEditState', () {
+    Map<String, dynamic> baseJson() => deepToJson(eg.streamMessage()) as Map<String, dynamic>;
+
+    test('edit history is absent', () {
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+      ).messageEditState).equals(MessageEditState.none);
+
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+          ..['last_edit_timestamp'] = 1678139636).messageEditState
+      ).equals(MessageEditState.edited);
+    });
+
+    test('edit history exists', () {
+      // Only channel name changed: moved
+      check(Message.fromJson(baseJson()
+        ..['edit_history'] = [{
+          'prev_stream': 'old_stream',
+          'stream': 'new_stream',
+        }]
+      ).messageEditState).equals(MessageEditState.moved);
+
+      // Only topic name changed: moved
+      check(Message.fromJson(baseJson()
+        ..['edit_history'] = [{
+          'prev_topic': 'old_topic',
+          'topic': 'new_topic',
+        }]
+      ).messageEditState).equals(MessageEditState.moved);
```

Comment:
These tests can be made more compact with a local helper like so:
```suggestion
      checkEditState(expected: MessageEditState.moved, [{
        'prev_topic': 'old_topic',
        'topic': 'new_topic',
      }]);
```

Making the test cases more compact like that is helpful in two ways:
* It makes them shorter vertically, so that one can see more of them at once.
* It cuts out elements that are repetitive from test case to test case, so that it's easier to focus on the parts that are interesting and different in each test case.

Both of those are helpful for scanning the list and thinking about what the tests are saying, and trying to think of other cases that aren't yet covered and would be interesting to test.